ReactiveObjC/RACNeverSignal.m

```diff
@@ -0,0 +1,67 @@
+//
+//  RACNeverSignal.m
+//  ReactiveObjC-macOS
+//
+//  Created by Yaron Inger on 19/12/2018.
+//  Copyright (c) 2018 GitHub, Inc. All rights reserved.
+//
+
+#import "RACNeverSignal.h"
+
+#import <ReactiveObjC/EXTScope.h>
+
+#import "RACCompoundDisposable.h"
+#import "RACPassthroughSubscriber.h"
+
+@interface RACNeverSignal ()
+
+/// Array that holds the subscribers to avoid them disposing the subscription.
+@property (nonatomic, readonly) NSMutableArray<id<RACSubscriber>> *subscribers;
+
+@end
+
+@implementation RACNeverSignal
+
+- (instancetype)init {
+  if (self = [super init]) {
+    _subscribers = [NSMutableArray array];
+  }
+  return self;
+}
+
++ (RACSignal *)never {
+  return [[self alloc] init];
+}
+
+- (NSString *)name {
+  return @"+never";
+}
+
+- (nullable RACDisposable *)subscribe:(id<RACSubscriber>)subscriber {
+  NSCParameterAssert(subscriber != nil);
+
+  NSMutableArray *subscribers = self.subscribers;
+  @synchronized (subscribers) {
+    [subscribers addObject:subscriber];
+  }
+
+  RACCompoundDisposable *disposable = [RACCompoundDisposable compoundDisposable];
```

Comment:
Why is this compound? as opposed to, say, a serial disposable or even the block disposable directly?

Reply:
I see you've answered the same question by Shachar. Fair enough then. It seems fine.

Re: why not use subject - 
I think this implementation is clearer and avoids potential problems with malicious (or more likely, stupid) downcasting.

Reply:
A `RACSubject` is also heavier in terms of objects allocated per signal.